lib/OpenQA/Git.pm

```diff
@@ -54,13 +54,11 @@ sub set_to_latest_master ($self, $args = undef) {
     }
 
     if (my $update_branch = $self->config->{update_branch}) {
-        my $res = run_cmd_with_log_return_error([@git, 'restore', '--staged', '.']);
-        return _format_git_error($res, 'Unable to restore staged files') unless $res->{status};
-        $res = run_cmd_with_log_return_error([@git, 'restore', '.']);
-        return _format_git_error($res, 'Unable to restore files') unless $res->{status};
-        $res = run_cmd_with_log_return_error([@git, 'clean', '-f']);
-        return _format_git_error($res, 'Unable to clean untracked files') unless $res->{status};
-        $res = run_cmd_with_log_return_error([@git, 'rebase', $update_branch]);
+        if (($self->config->{do_cleanup} || '') eq 'yes') {
```

Comment:
The default specified in `Setup.pm` should be effective so there's no need for `|| ''`:
```suggestion
        if (($self->config->{do_cleanup}) eq 'yes') {
```